pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -80,6 +82,8 @@ public class SegmentStatusChecker extends ControllerPeriodicTask<SegmentStatusCh
 
   private long _lastDisabledTableLogTimestamp = 0;
 
+  private ServerInfoFetcher _serverInfoFetcher;
```

Comment:
Let's not have this as a class member. The reason is the cache you have in the class is not cleared. If you create an `ServerInfoFetcher` object in each run of the segment status checker, there's no need to worry about the cache eviction.